fix: navigation respects basepath when no `to` was given

jvllmr opened the pull request:
Simple fix by passing "." to `Router.resolvePathWithBase` instead of using `fromPath`. Parameterized relative URL tests with basepath to avoid future regressions. I also managed to reproduce the issue this way.

fixes #4401 
fixes #4471

schiller-manuel commented:
thanks for your PR!

I am wondering why this works, given that this user wrote:

https://github.com/TanStack/router/issues/4401#issuecomment-2976060849
<img width="971" alt="Screenshot 2025-06-21 at 01 05 50" src="https://github.com/user-attachments/assets/25d2d5c0-b9aa-4303-aff9-afcbe68e20d0" />

SeanCassiere reviewed (changes requested):
This does look OK to me.

@schiller-manuel’s point is interesting. Worth looking into.

@jvllmr please:

- Replicate the tests in the Solid package
- Use unique test names for each run: `test name xyz for $basepath`

jvllmr commented:
> This does look OK to me.
> 
> @schiller-manuel’s point is interesting. Worth looking into.
> 
> @jvllmr please:
> 
> * Replicate the tests in the Solid package
> * Use unique test names for each run: `test name xyz for $basepath`

I wondered about that case as well, but I focused on reproducing the problem with tests and started with the clue from the original issue. I could add the case from the issue to the tests as well just to be sure.

I made the test names unique and started on porting the tests to solid as well and pushed my progress for now because I ran out of time for today.

SeanCassiere commented:
> I could add the case from the issue to the tests as well just to be sure.

Yes please. Better to cover our bases with more testing to cover the case.

jvllmr commented:
All done ✅ 

- I have finished porting the tests to `@tanstack/solidrouter`. When doing that I found an inconsistency with resolving the `from` option between `react-router` and `solid-router`. I updated `solid-router` to match the behavior of `react-router`
- I added a new testcase for the `Link` component that should reproduce the problem from the issue comment. However, when I reverted my patch the test still passed. Maybe there is more context to that we do not know or it is a false positive altogether?

SeanCassiere reviewed (approved):
> I have finished porting the tests to `@tanstack/solid-router`. When doing so I found an inconsistency with resolving the `from` option between `react-router` and `solid-router`. I updated `solid-router` to match the behavior of `react-router`.

👍🏼

> I added a new testcase for the `Link` component that should reproduce the problem from the issue comment. However, when I reverted my patch the test still passed. Maybe there is more context to it we do not know or it is a false positive altogether?

🤔... I'm starting to think that comment is a red herring.

Running the original sandbox (from the mentioned issue) against the pr.pkg.new packages from this PR, the runtime issue of navigation without `to` resulting in the omission of the configured basepath, does seem to be fixed.

---

Either ways, this was clearly an issue in the resolution logic not accounting for adding the basepath when `to` wasn't supplied.

If any further fixes are required, we'll fix forward.

SeanCassiere commented:
@jvllmr thank you for taking up this one and working with us on fixing this problem!
